Widgets 2.0 (#4028)
DataDog/documentation: content/en/graphing/widgets/heat_map.md

---
title: Heat Map Widget
kind: documentation
description: Build temporal heat map over a given metric.
further_reading:
- link: "graphing/dashboards/timeboard/"
  tag: "Documentation"
  text: "Timeboards"
- link: "graphing/dashboards/screenboard/"
  tag: "Documentation"
  text: "Screenboard"
---

> The heat map visualization is great for showing metrics aggregated across many tags, such as *hosts*. The more hosts that have a particular value, the darker that square is.

Reviewer: ```suggestion
The heat map visualization shows metrics aggregated across many tags, such as *hosts*. The more hosts that have a particular value, the darker that square is.
```

This visualization displays only a single metric query; additional queries are disregarded.

**Note**: Outlier detection cannot be performed for this visualization.

{{< img src="graphing/widgets/heat_map/heat_map.png" alt="Heat Map" responsive="true">}}

## Setup

{{< img src="graphing/widgets/heat_map/heat_map_setup.png" alt="Heat Map setup" responsive="true" style="width:80%;">}}

### Configuration

> Configure your metric query as usual. Note that this visualization type is useful only when metrics are aggregated across Tag Keys , e.g. for each `host`.

Reviewer: ```suggestion
Configure your metric query as usual. Note that this visualization type is useful only when metrics are aggregated across tag keys , e.g. for each `host`.
```

> Make a selection in the "`avg`/`max`/`min`/`sum by`…" control to see your data across the associated tags.

Reviewer: ```suggestion
Make a selection in the "`avg`/`max`/`min`/`sum by`/etc." control to see your data across the associated tags.
```

### Options
#### Event Overlay

Add events from related systems to add more context to your graph. For example, you can add GitHub commits, Jenkins deploys, or Docker creation events. Expand the **Event Overlays** section and enter a query to display those events. Use the same query format as for the [Event Stream][1], for example:

| Query                       | Description                                                |
|-----------------------------|------------------------------------------------------------|
| `sources:jenkins`           | Shows all events from the Jenkins source.                  |
| `tag:role:web`              | Shows all events with the tag `role:web`.                  |
| `tags:$<TEMPLATE_VARIABLE>` | Shows all events from the selected [Template Variable][2]. |

#### Y-axis controls

Y-axis controls are available via the UI and the JSON editor. They allow you to:

* Clip the y-axis to specific ranges.

> * Remove outliers either by specifying a percentage or an absolute value to remove outliers.

Reviewer: Let's specify how this percentage or absolute value removes outliers.
